WooCommerce/Classes/ViewRelated/Products/Edit Product/Linked Products/LinkedProductsViewController.swift

```diff
@@ -142,16 +146,57 @@ private extension LinkedProductsViewController {
     }
 
     func configureUpsellsButton(cell: ButtonTableViewCell) {
-        let buttonTitle = Localization.buttonTitle(count: viewModel.upsellIDs.count)
-        cell.configure(style: .secondary, title: buttonTitle) {
+        guard let product = product as? EditableProductModel else {
+            return
+        }
+
+        // TODO: add analytics M5 https://github.com/woocommerce/woocommerce-ios/issues/3151
+        let viewConfiguration = LinkedProductsListSelectorViewController.ViewConfiguration(title: Localization.titleScreenAddUpsellProducts,
+                                                                                           addButtonEvent: .groupedProductLinkedProductsAddButtonTapped,
+                                                                                           productsAddedEvent: .groupedProductLinkedProductsAdded,
+                                                                                           doneButtonTappedEvent: .groupedProductLinkedProductsDoneButtonTapped,
+                                                                                           deleteButtonTappedEvent:
+                                                                                            .groupedProductLinkedProductsDeleteButtonTapped)
+
+        let viewController = LinkedProductsListSelectorViewController(product: product.product,
+                                                                      linkedProductIDs: viewModel.upsellIDs,
+                                                                      viewConfiguration: viewConfiguration) { [weak self] upsellIDs in
+            self?.viewModel.handleUpsellIDsChange(upsellIDs)
+            self?.tableView.reloadData()
+            self?.navigationController?.popViewController(animated: true)
+        }
+
 
+        let buttonTitle = Localization.buttonTitle(count: viewModel.upsellIDs.count)
+        cell.configure(style: .secondary, title: buttonTitle) { [weak self] in
+            self?.show(viewController, sender: self)
         }
     }
 
     func configureCrossSellsButton(cell: ButtonTableViewCell) {
-        let buttonTitle = Localization.buttonTitle(count: viewModel.crossSellIDs.count)
-        cell.configure(style: .secondary, title: buttonTitle) {
+        guard let product = product as? EditableProductModel else {
```

Comment:
This is true, for the moment they are similar, but after this task https://github.com/woocommerce/woocommerce-ios/issues/3151, and after adding all the missing events, they will be different since we need to pass different events inside the `ViewConfiguration`, different button titles, different ViewModel handling for updating the different values.

Reply:
Makes sense!